### PR TITLE
Use indexing instead of masked_select for faster normalization

### LINKED ISSUE #1017: `masked_select` on Z-Normalization slower than indexing
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Bug summary

The torch function `masked_select` is slower than indexing:

```
In [1]: import torch

In [2]: x = torch.rand(300, 300, 300)

In [3]: y = torch.rand(300, 300, 300) > 0.5

In [4]: %timeit -n 20 x[y]
102 ms ± 2.33 ms per loop (mean ± std. dev. of 7 runs, 20 loops each)

In [5]: %timeit -n 20 x.masked_select(y)
152 ms ± 2.17 ms per loop (mean ± std. dev. of 7 runs, 20 loops each)
```

### Code for reproduction

```python
In [1]: import torch

In [2]: x = torch.rand(300, 300, 300)

In [3]: y = torch.rand(300, 300, 300) > 0.5

In [4]: %timeit -n 20 x[y]
102 ms ± 2.33 ms per loop (mean ± std. dev. of 7 runs, 20 loops each)

In [5]: %timeit -n 20 x.masked_select(y)
152 ms ± 2.17 ms per loop (mean ± std. dev. of 7 runs, 20 loops each)
```

### Actual outcome

Faster indexing with `x[y]`

### Error messages

_No response_

### Expected outcome

Faster normalization.

### System info

```shell
Platform:   macOS-13.1-x86_64-i386-64bit
TorchIO:    0.18.86
PyTorch:    1.13.1
SimpleITK:  2.2.1 (ITK 5.3)
NumPy:      1.24.1
Python:     3.9.13 (main, Jul 27 2022, 15:28:09)
[Clang 13.0.0 (clang-1300.0.29.30)]
```

### PR DESCRIPTION
**Pull request by ramonemiliani93**
Fixes #1017.

**Description**
Uses indexing instead of `masked_select` for faster normalization.

**Checklist**

- [x] I have read the [`CONTRIBUTING`](https://github.com/fepegar/torchio/blob/main/CONTRIBUTING.rst) docs and have a developer setup (especially important are `pre-commit`and `pytest`)
- [x] Non-breaking change (would not break existing functionality)
- [ ] Breaking change (would cause existing functionality to change)
- [ ] Tests added or modified to cover the changes
- [x] Integration tests passed locally by running `pytest`
- [ ] In-line docstrings updated
- [ ] Documentation updated, tested running `make html` inside the `docs/` folder
- [x] This pull request is ready to be reviewed
- [ ] If the PR is ready and there are multiple commits, I have [squashed them and force-pushed](https://www.w3docs.com/snippets/git/how-to-combine-multiple-commits-into-one-with-3-steps.html#force-pushing-commits-7)

### CODE REVIEW

**Comment by fepegar** [comment]
Thanks, @ramonemiliani93. Could you run some benchmarks as a sanity check? It looks like at some point `masked_select` seemed faster:
- https://github.com/fepegar/torchio/issues/202

**Comment by fepegar** [comment]
I've just seen your tests in #1017, sorry. I guess that's good enough! Maybe they've changed something in PyTorch in the meantime.
